pages/docs/tutorials/native/gradle-for-kotlin-native.md

```diff
@@ -0,0 +1,161 @@
+
+Kotlin/Native has a plugin for [Gradle](https://gradle.org) which allows us to compile, build libraries, reference libraries, and in general perform all the actions that we would need to build our application. 
+
+In this tutorial we'll see how to
+
+* [Build an application with Gradle](#building-an-application)
+* [Build and reference a library with Gradle](#building-and-referencing-a-library)
+* [Generate CMake files to use with CLion](#generating-cmake-files)
+
+
+**Note**: This tutorial assumes somewhat familiarity with Gradle. If you're new to Gradle, while you should be able to follow along, it would be beneficial to review some of the 
+[Getting Started guides](https://gradle.org/guides/#getting-started). To install Gradle on your system please see the [Install guide](https://gradle.org/install/).
+
+
+## Building an Application
+
+The following Gradle script (named `build.gradle`) is the simplest needed to compile an application 
+
+```groovy
+buildscript {
+    repositories {
+        mavenCentral()
+        maven {
+            url "https://dl.bintray.com/jetbrains/kotlin-native-dependencies"
+        }
+    }
+    dependencies {
+        classpath "org.jetbrains.kotlin:kotlin-native-gradle-plugin:+"
+    }
+}
+
+apply plugin: 'konan'
+
+konan.targets = ['macbook']
+
+konanArtifacts {
+    program('hello')
+}
+```
+
+The `buildscript` section defines the repositories and dependencies required for the Gradle plugin for Kotlin/Native which is `konan`. Then comes the actual parameters for compiling:
+
+1. The `konan.targets` can contain one or more target platforms.
+
+2. The `konanArtifacts` contains the actual instructions telling Gradle what to build. In our case we're saying we want to build an 
+executable called `hello`. 
+
+An important aspect for this build script to work is that we follow the default convention of where our source
+files are located, which is `src/main/kotlin`. We can of course change this behaviour by defining the `srcFiles`, which we'll see further down. 
+
+We can now create a Kotlin file with the following contents and place it in the `src/main/kotlin` folder:
+
+```kotlin
+fun main(args: Array<String>) {
+    println("Hello Gradle!")
+}
+``` 
+
+To recap, this should be our directory structure layout:
+
+![Directory Structure Before Build]({{ url_for('tutorial_img', filename='native/gradle-for-kotlin-native/dir-before-build.png')}}))
+
+
+From the root directory, where our `build.gradle` file is, we can now run the following command:
+
+    gradle build
+    
+    
+which will build the application. The end result should be the following:
+
+![Directory Structure After Build]({{ url_for('tutorial_img', filename='native/gradle-for-kotlin-native/dir-after-build.png')}}))
+     
+
+We can see that there is a new `build` directory with a subdirectory for each target. Inside this folder we have the actual executable with the name
```

Comment:
You use "folder" and "directory" interchangeably. Better to choose one term and stick to it.